src/components/Menu/MenuItem.tsx

```diff
@@ -179,6 +297,55 @@ class MenuItem extends UIComponent<Extendable<MenuItemProps>, MenuItemState> {
 
     _.invoke(this.props, 'onFocus', e, this.props)
   }
+
+  private closeAllMenus = e => {
```

Comment:
it is hard to infer the difference of logic provided by this method vs subsequent `closeMenuAndFocusNextParentItem` one by using their names only - more than that, quite a lot of implementation code is similar. Maybe we could refactor it to be a single method with some parameter that will drive (and describe) this difference?

Reply:
It was initially one method, but we have different handlings if the esc key was used, or right arrow, so that suggestions from behaviors point of view was that we should have different methods for these, as we don't want to use the check for the keys inside the component's logic.